### ISSUE TITLE
Create Github Pages Project for "Topcoder Cookbook"

### ISSUE THREAD

**Issue by hokienick** [topcoderinc/cab#78]
Forum Post where this was mentioned:
https://apps.topcoder.com/forums/?module=Thread&threadID=902490&start=0

"Topcoder Cookbook":
https://weltam.wordpress.com/topcoder-cookbook/

There was a page that was created that contains links to various forum posts that either help new members get acclimated to the platform or assist in solving or thinking about certain algorithms.

Members would like to have an open-sourced platform that makes it easily readable with this information and ability to add more later.

**Comment by tcNickolas** [comment]
These are not just "various forum posts", this was a draft of a book which Topcoder was going to crowdsource and publish with O'Reilly in their Cookbook series, but it was never finished.

My actual suggestion was broader: port the actual articles (recipes) from TC forums to github pages, not just maintain the list of recipes. This has a whole lot of advantages: it allows to have images, tables and other fancy formatting (might be a bit tricky to figure out formulas but possible), and it removes dependency from forums (no spamming, no forums being down etc).

You can see what we've done in https://github.com/e-maxx-eng/e-maxx-eng/ , it's a similar project

**Comment by hokienick** [comment]
@tcNickolas - If I started a repo, would you like to manage it?

**Comment by tcNickolas** [comment]
I can help with this a bit (for example, I have cleaned-up versions of several chapters in Google Docs which are a good start). But if it is to be an open-source project, long-term it needs several maintainers, I can't be the only one.

**Comment by hokienick** [comment]
@tcNickolas Agreed.  Maybe you can help start it off and then we can find others who would be interested in helping maintain.

**Comment by hokienick** [comment]
@tcNickolas - the team met on 06/28 and felt that the best location for this information would be in our Help Center.  If we were to give you access to this, would you care to start off the "cookbook" and then we will find someone to help out?

**Comment by tcNickolas** [comment]
@hokienick Topcoder Help Center is not an open source platform, thus it has a lot of flaws.

* It has very low visibility, and any material placed there will be hard to find. It is already a huge pain to find a lot of things on Topcoder website if they are older than a couple of months, and I don't see this improving.
* The contents there can be deleted at any time for no reason without any notification (we've already seen this happen with Cookbook forums once, and they had to be recovered manually with a lot of effort). A Github repo can be cloned and preserved, so it is only deleted if the whole Github closes.
* It requires giving access separately to everybody who wants to volunteer contributing, thus much fewer people will want to go through the effort (creating a Github account is much easier). This kind of project can't be done by a single person or even a couple of people.
* It doesn't showcase the contributions of individuals the way Github does, and doesn't have rewards systems like CodeCuriosity and Hacktoberfest already associated with it.
* Finally, the really useful recipes in Cookbook are the ones not limited to Topcoder; it has not just topics introducing different formats of Topcoder contests but awesome material on topics like dynamic programming which don't belong to Topcoder Help Center.

I don't know how many of these points "the team" has considered before making this decision. I don't think it's a great location for this project.

**Comment by tcNickolas** [comment]
@hokienick One more reason:
Topcoder platform itself is flaky: even if the content is there, it might not be displayed for an arbitrary reason. Take a look at the three most useful recipes I've seen in the Cookbook:
https://apps.topcoder.com/forums/?module=Thread&start=0&threadID=700080
https://apps.topcoder.com/forums/?module=Thread&start=0&threadID=697925
https://apps.topcoder.com/forums/?module=Thread&start=0&threadID=697369
They all used to have great content, now they just show as empty even though they are not deleted. This problem was an issue in the forums for a while and it is still not fixed. The content on Github is always displayed as long as Github itself is up.

**Comment by agorbunovs** [comment]
@hokienick created a repo: https://github.com/topcoderinc/cookbook
Everyone is welcome o contribute (after the initial structure is introduced by me and @tcNickolas).

**Comment by RodionGork** [comment]
Friends, @hokienick, @agorbunovs Hi!

I was recently reached by Masha aka @tcNickolas and according to her suggestions I'm ready to help a bit with an app which renders the github content as a free web-site in google appengine (just as https://e-maxx-eng.appspot.com is rendered from https://github.com/e-maxx-eng/e-maxx-eng/).

Sorry for delay as it take me some time to make few improvements so that you can reuse the code. So now I'm ready to provide instructions on how to deploy it (actually sent a part of it to Masha by email)...

Shortly speaking we (or you) need:

1. Create a new appengine project in google cloud console
2. Either invite me temporarily to it or upload the code there and add a config with github particulars
3. Slightly change your github project structure (rename `md` folder to `src` and create `index.md` in it to work as a landing page - probably also create `src/_templates/default.html` - feel free to borrow it first and modify later).

I think that's all... Though probably later it would be good if I externalize css files so that style of pages can be easily changed (without it styles still can be affected in template, I think).
